runtime/interpreter/value.go

```diff
@@ -4874,12 +4920,18 @@ func (v *DictionaryValue) Count() int {
 }
 
 // TODO: unset owner?
-func (v *DictionaryValue) Remove(keyValue Value) (existingValue Value) {
+func (v *DictionaryValue) Remove(inter *Interpreter, keyValue Value) (existingValue Value) {
 	v.modified = true
 
 	key := dictionaryKey(keyValue)
 	existingValue, exists := v.Entries[key]
 
+	if v.prevDeferredKeys != nil {
+		if storageKey, ok := v.prevDeferredKeys[key]; ok {
```

Comment:
So if a resource is moved out of the dictionary, we delete its old key in storage, and then rely on resource semantics to make sure it is stored or destroyed later?

Reply:
Correct, well put. I'll add that as a comment :-)

Reply:
Improved documentation a bit in f594c0ff341ce234fd31c4e06012596ec2984c6c